scripts/add_repo.sh

```diff
@@ -1,98 +1,135 @@
-#!/bin/bash -x
+#!/bin/bash
 
 set -eu
```

Comment:
Real minor nit:

I personally prefer seeing the option rather than the flag:

```
set -o errexit
set -o nounset
```

You may also want a `set -o pipefail` but I need to read through the rest of the changes to see if that's applicable.